build: update angular version of in-memory-web-api

crisbeto opened the pull request:
Changes the required version of Angular in `angular-in-memory-web-api` to version 14.0.0. 

Fixes #46332.

pkozlowski-opensource commented on packages/misc/angular-in-memory-web-api/package.json:
```diff
@@ -1,17 +1,14 @@
-    "@angular/core": "^13.0.0",
-    "@angular/common": "^13.0.0",
+    "@angular/core": ">=14.0.0",
+    "@angular/common": ">=14.0.0",
```
Hmm, while I agree that the dependency on the Angular's public API surface is limited we really can't exclude breaking changes. So I'm not sure I'm that comfortable with doing  `>=14`. Let's discuss more during standup?

crisbeto replied:
Just to capture what we discussed in DM: this is a package we keep forgetting to update and it hasn't been broken for a long time. I think that we can get away with this version range, because we run its tests against the latest code on the `main` branch so we'll find out if we break it by accident.

alan-agius4 replied:
For the record, I agree with @pkozlowski-opensource. `>=14` is too broad.

alan-agius4 replied:
> Just to capture what we discussed in DM: this is a package we keep forgetting to update and it hasn't been broken for a long time. I think that we can get away with this version range, because we run its tests against the latest code on the `main` branch so we'll find out if we break it by accident.

But on the main branch, we don't test the shipped package from NPM.  Officially Angular libraries are only 2 major versions forward compatible. Example a library built using NGC version 13 is only guaranteed to word with version 13, 14 and 15.  This might send a mixed signal to the community.

alan-agius4 replied:
If we keep forgetting to release this package, we should update our release docs and not wide the peer dependencies range.

pkozlowski-opensource replied:
Update: discussed this during the framework standup and leaning towards:
* change the PR to constraints peer dependencies to `^14.0.0` (pretty much what we did for previous major releases);
* discuss the future of the `in-memory-web-api` package (its location and the release process) during the next framework sync.

alan-agius4 commented on packages/misc/angular-in-memory-web-api/package.json:
```diff
@@ -1,17 +1,14 @@
+    "@angular/common": ">=14.0.0",
     "rxjs": "^6.5.3 || ^7.4.0"
   },
-  "dependencies": {
```
Why was `tslib` removed from `dependencies`? This is still required.

crisbeto replied:
Isn't it going to be brought in by `@angular/core`?

alan-agius4 replied:
That is relying on module hoisting which is not correct.

Also, the `tslib` version needed for this package might different from that of `@angular/core` example when `@angular/core` is compiled using a different version of TypeScript. 

Every library that is compiled with TypeScript and doesn't inline tslib should have such library as a direct dependency.

crisbeto committed:
build: update angular version of in-memory-web-api

Changes the required version of Angular in `angular-in-memory-web-api` to version 14.0.0.

Fixes #46332.

crisbeto commented:
I've addressed the feedback.

pkozlowski-opensource reviewed (approved):
LGTM

alan-agius4 reviewed (approved):
LGTM

jessicajaniuk commented:
This PR was merged into the repository by commit d6880c92f9236be9d232d361d6e2e8eaf09feeb9.
